app/src/main/java/homer/model/lock/SimpleLock.java

```diff
@@ -0,0 +1,39 @@
+package homer.model.lock;
+
+import homer.DeviceInfoImpl;
+import homer.api.DeviceIdImpl;
+import homer.api.DeviceInfo;
+import homer.api.ToggleableDevice;
+
+/**
+ * Implementation of a {@link ToggleableDevice} representing a lock.
+ */
+public final class SimpleLock implements ToggleableDevice<Boolean> {
+
+    /**
+     * Type of device.
+     */
+    public static final String DEVICE_TYPE = "SimpleLock";
+    private boolean isLocked = false;
+
+    @Override
+    public DeviceInfo getInfo() {
+        return new DeviceInfoImpl(new DeviceIdImpl(), DEVICE_TYPE);
```

Comment:
I think this should be done in the controller. E.g.
```
class myClass implements ToggleableDevice<something> {
  private DeviceInfo info
  // something else
  public myClass(DeviceInfo info, ... ) {
      this.info = info;
      ...
   }

   public DeviceInfo getDeviceInfo(){
       return this.info
   }

  ...
```

When we're instantiating the Device object we'll intantiate a DeviceInfo object with the info:
```
DeviceInfo myBeautifulInfos = new DeviceInfoImpl(something);
Device myBeautifulDevice = new myClass(myBeautifulInfo, ...);
```

Reply:
Or:
```
Device myBeautifulDevice = new myClass(new DeviceInfoImpl(something), ...);
```

Reply:
While I agree with the id being created by the controller, I feel like the type should be reported by the device.
Either way, if the controller created both the id and the type, there would be no point in storing this info in the device at all